streams/src/main/java/org/apache/kafka/streams/processor/Punctuator.java

```diff
@@ -21,6 +21,10 @@
  */
 public interface Punctuator {
 
+    /**
+     * Perform the scheduled periodic operation
+     *
+     * @param timestamp the stream time when the operation is being called
```

Comment:
AFAIK this should really be the stream time when the operation is being called when running in `{@link PunctuationType#STREAM_TIME}` and the current system time when the operation called when running `{@link PunctuationType#WALL_CLOCK_TIME}`

Reply:
I disagree. Users can get the current wall-clock-time by calling `System.currentTimeMillis()` -- but there is no change to access current "stream time" --  passing in "stream time" is the only way to make this information available.

Reply:
Ah I see! How about pointing this out in the comment?

Reply:
Why don’t we call it stream time instead of time stamp, if it’s behaiviour doesn’t change?

Reply:
Actually the passed in `timestamp` does depend on the punctuationType (see `maybePunctuateXXTime` in `StreamTask`). Will update accordingly.

Reply:
Ack. Missed that part...